packages/shared/lib/clients/oauth2.client.ts

```diff
@@ -36,25 +36,22 @@ export function getSimpleOAuth2ClientConfig(
             authorizeHost: authorizeUrl.origin,
             authorizePath: authorizeUrl.pathname
         },
-        http: {
-            headers,
-            // @ts-expect-error badly documented feature https://github.com/hapijs/wreck/blob/ba28b0420d6b0998cd8e61be7f3f8822129c88fe/lib/index.js#L34-L40
-            agents:
-                httpAgent && httpsAgent
-                    ? {
-                          http: httpAgent,
-                          https: httpsAgent,
-                          httpsAllowUnauthorized: httpsAgent
-                      }
-                    : undefined
-        },
         options: {
             authorizationMethod: authConfig.authorization_method || 'body',
             bodyFormat: authConfig.body_format || 'form',
             // @ts-expect-error seems unused ?
             scopeSeparator: provider.scope_separator || ' '
         }
     };
+
+    // define getters for agents so that they aren't clone within simple-oauth2
+    const httpConfig = { headers, agents: {} };
+    Object.defineProperty(httpConfig.agents, 'http', { get: () => httpAgent, enumerable: true });
+    Object.defineProperty(httpConfig.agents, 'https', { get: () => httpsAgent, enumerable: true });
+    Object.defineProperty(httpConfig.agents, 'httpsAllowUnauthorized', { get: () => httpsAgent, enumerable: true });
```

Comment:
I think we can simplify and avoid having to use `defineProperty` like this:
```
const getAgentConfig = (): { agents: AgentConfig } => ({
  agents: {
    get http() { return httpAgent; },
    get https() { return httpsAgent; },
    get httpsAllowUnauthorized() { return httpsAgent; }
  }
});

const clientConfig: Merge<ModuleOptions, { http: WreckHttpOptions }> = {
    ...,
    http: {
      headers,
      ...getAgentConfig()
    }
  };
```
And adding a comment to mention the getter dynamic lookup trick used here. I am sure next week I am gonna have forgotten about it already

Reply:
I had already hit merge but I'll put a follow-up in real quick and merge it if this works out for that hot spot in the profiles, otherwise we'll just revert.

Reply:
https://github.com/NangoHQ/nango/pull/3457